url888b6fd9b4_jmacglashan_commandsToTasks_tgz-pJ8-commands_tests_Model3TestsJ8/src/domain/singleagent/sokoban/SokobanDomain.java

```diff
@@ -0,0 +1,1345 @@
+
+	private static SADomain 			SOKOBANDOMAIN = null;
+
+	public static void main(String args[]){
+
+		//command line arguments
+		int vLocation = -1;
+		int fLocation = -1;
+		int tLocation = -1;
+		int cLocation = -1;
+		int dLocation = -1;
+		int wLocation = -1;
+		int sLocation = -1;
+
+
+		if (args.length == 0 || args[0].equals("-help")){
+			System.out.println("Usage: SokobanDomain [-c commandsfile] [-t] [-v] [-f statefile]");
+			System.out.println("-c runs a set of commands from a text file");
+			System.out.println("-d runs a set of commands from a directory of text files");
+			System.out.println("-t runs the terminal domain explorer");
+			System.out.println("-v runs the visual domain explorer");
+			System.out.println("-f will load a starting state from a text file rather than using the default one");
+			System.out.println("-w will prevent the creation of walls (warning: this may have unintented consequences)");
+			System.exit(0);
+		}
+
+		//check for all flags
+		for (int i = 0; i < args.length; i++){
+			if (args[i].equals("-v") && vLocation == -1){
+				vLocation = i;
+			}
+			else if (args[i].equals("-v")) {
+				System.out.println("Error: duplicate flag \"-v\".");
+				System.exit(1);
+			}
+			if (args[i].equals("-w") && wLocation == -1){
+				wLocation = i;
+			}
+			else if (args[i].equals("-w")) {
+				System.out.println("Error: duplicate flag \"-w\".");
+				System.exit(1);
+			}
+			if (args[i].equals("-t") && tLocation == -1){	
+				tLocation = i; 
+			}
+			else if (args[i].equals("-t") && tLocation != -1) {
+				System.out.println("Error: duplicate flag \"-t\".");
+				System.exit(1);
+			}
+			if (args[i].equals("-f") && fLocation == -1){
+				fLocation = i;
+			}
+			else if (args[i].equals("-f")) {
+				System.out.println("Error: duplicate flag \"-f\".");
+				System.exit(1);
+			}
+			if (args[i].equals("-c") && cLocation == -1){
+				cLocation = i;
+			}
+			else if (args[i].equals("-c")) {
+				System.out.println("Error: duplicate flag \"-c\".");
+				System.exit(1);
+			}
+			if (args[i].equals("-d") && dLocation == -1){
+				dLocation = i;
+			}
+			else if (args[i].equals("-d")) {
+				System.out.println("Error: duplicate flag \"-d\".");
+				System.exit(1);
+			}
+			if (args[i].equals("-s") && sLocation == -1){	
+				sLocation = i; 
+			}
+			else if (args[i].equals("-s") && sLocation != -1) {
+				System.out.println("Error: duplicate flag \"-s\".");
+				System.exit(1);
+			}
+
+		}
+
+		if (sLocation != -1){
+			MAXX = Integer.parseInt(args[sLocation + 1]);
+			MAXY = Integer.parseInt(args[sLocation + 2]);
+		}
+		else {
+			MAXX = 24;
+			MAXY = 24;
+		}
+
+		MAP = new int[MAXX+1][MAXY+1];
+		SokobanDomain constructor = new SokobanDomain();
+		Domain domain = constructor.generateDomain();
+
+		State st = constructor.getCleanState();
+
+
+		if (fLocation != -1){
+			BufferedReader in = null;
+			String newState = null;
+			try {
+				in = new BufferedReader(new FileReader(args[fLocation + 1]));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *in*. The resource is referenced at lines 202, 208. The resource is closed at line 209. There are other execution paths that do not contain closure statements, for example, when BufferedReader.readLine throws an exception. Ensure *in* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).